precompile/allowlist/allowlist.go

```diff
@@ -119,7 +117,15 @@ func createAllowListRoleSetter(precompileAddr common.Address, role Role) contrac
 
 		// Verify that the caller is an admin with permission to modify the allow list
 		callerStatus := GetAllowListStatus(stateDB, precompileAddr, callerAddr)
-		if !callerStatus.IsAdmin() {
+		if callerStatus == ManagerRole {
+			// Get current status.
+			// Before the manager role, we never checked the status of the address we are trying to modify.
+			// So we should keep the same behaviour by special casing this.
+			modifyStatus := GetAllowListStatus(stateDB, precompileAddr, modifyAddress)
+			if !callerStatus.CanModify(modifyStatus, role) {
+				return nil, remainingGas, fmt.Errorf("%w: modify address: %s, from role: %s, to role: %s", ErrManagerCannotModify, callerAddr, modifyStatus, role)
+			}
+		} else if !callerStatus.IsAdmin() {
```

Comment:
Hmm good point. Performing the extra read operation doesn't change the result if we don't use it, so this should be fine.

Another question here is whether we should charge for the extra read operation. If we do, then we need to keep this separation to charge gas for the extra read operation only when it occurs.

I'd be in favor of keeping the gas cost the same and switching to always use `CanModify` to keep the code as simple as possible. Otherwise, introducing `CanModify` to cover one case but not actually using it when the `callerStatus` is `Admin` makes this code more confusing to reason about what invariants we need to maintain vs. how they're actually used imo.

What do you think?

Reply:
I agree with keeping the gas cost the same. If read operation doesn't matter that much I agree with keeping the code simple and clean.